Automattic/wp-calypso
# Site Profiler: Add icon and title when the site is hosted in WP.com

Resolves https://github.com/Automattic/dotcom-forge/issues/7281

## Proposed Changes

* Add Icon and domain title message to the results header component based on if the site is hosted on WordPress.com

Figma design 1: P7Jjk4nbUTahUk8aHkG6NQ-fi-712-25225
Figma design 2: P7Jjk4nbUTahUk8aHkG6NQ-fi-712-25704


|WPCom site | Non WPCom site|
|-------|------|
|![CleanShot 2024-05-24 at 15 32 19@2x](https://github.com/Automattic/wp-calypso/assets/3519124/a2e9f5e1-15eb-4eae-85dd-fe1cd89ea0b8)| ![CleanShot 2024-05-24 at 15 32 06@2x](https://github.com/Automattic/wp-calypso/assets/3519124/07933ee8-ffa2-4ea1-b549-db612acb5c0d)|


## Why are these changes being made?


* To indicate to the user that the site is hosted in WordPress.com according to the design

## Testing Instructions



* Apply this patch and in your local, navigate to the site profiler with a WordPress.com site, e.g. `/site-profiler/wordpress.com`
* Check the icon and domain title corresponds to the screenshot above for WordPress.com sites
* Navigate to a non WordPress.com site, e.g. `/site-profiler/example.com`
* Check the icon and domain title corresponds to the screenshot above for non WordPress.com sites

## Pre-merge Checklist



~~- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?~~
~~- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?~~
